### DISCUSSION TITLE
plugin works with FF, struggles with Chrome and Edge and shows 500 internal server error

### DISCUSSION THREAD

**Discussion by wkulesza** [OpenIDC/mod_auth_openidc#699] [Q&A] [answered]
Hi, great plugin for apache.
When we configured OIDC (using auth0 user-password-authentication) in virtual host for subdomain apache, when entering the required URL, we're brought to auth0 login. After providing correct credentials, user is brought to desired website (served by Tomcat through Docker on that server). This works on FF, but struggles on Chrome and Edge.

After providing credentials, instead of being brough to desired website (as with FF), 500 internal server error appears and url shows:

http://mysubdomain.mydomain.com/protected?code=some_string_here&state=other_string_here

obviously replaced correct strings above and subdomains/domains.

error log for apache is growing enourmously and includes entries like that:
[Thu Oct 07 09:07:36.798493 2021] [auth_openidc:error] [pid 1053947] [client 46.238.121.112:50330] oidc_restore_proto_state: no "mod_auth_openidc_state_B_UK_5JnZdtadBWWnFcIuRaiOqA" state cookie found
[Thu Oct 07 09:07:36.798679 2021] [auth_openidc:error] [pid 1053947] [client 46.238.121.112:50330] oidc_unsolicited_proto_state: could not parse JWT from state: invalid unsolicited response: [src/jose.c:754: oidc_jwt_parse]: cjose_jws_import failed: invalid argument [file: jws.c, function: cjose_jws_import, line: 781]
[Thu Oct 07 09:07:36.798688 2021] [auth_openidc:error] [pid 1053947] [client 46.238.121.112:50330] oidc_authorization_response_match_state: unable to restore state
[Thu Oct 07 09:07:36.798694 2021] [auth_openidc:error] [pid 1053947] [client 46.238.121.112:50330] oidc_handle_authorization_response: invalid authorization response state and no default SSO URL is set, sending an error...

[Thu Oct 07 08:01:01.308969 2021] [authz_core:debug] [pid 856536] mod_authz_core.c(817): [client 77.65.26.67:28873] AH01626: authorization result of Require valid-user : denied (no authenticated user yet), referer:

Guys, any suggestions on how to tackle this problem ?
I guess it has to do with cookies. This app doesn't have a logout functionality - is there a way to auto-logout user if there's inactivity or after some time? or create a logout page where he can go and logout/clear cookies etc ?

**Comment by zandbelt** [comment]
that is most probably due to changed `SameSite` cookie behaviours in browsers; upgrading to the latest version should fix this; session invalidation after inactivity is implemented, the interval can be changed in the config with `OIDCSessionInactivityTimeout`

**Comment by wkulesza** [comment, marked as answer]
@zandbelt Hi, thanks for pointers.
I noticed that on one server Edge allows me to login and on the other not - so assumption there are two different versions of this module could be correct.
How to check what version of that module I'm running? One server (running) is Debian 10, one (not allowing to login) is Ubuntu LTS.

I tried using:
```
strings /usr/lib/apache2/modules/mod_auth_openidc.so | grep mod_auth_openidc
```

and strangely on the debian server that works ok it prints: OpenID Connect Apache Module (mod_auth_openidc)
mod_auth_openidc-2.3.3

and on the ubuntu server that doesnt' work it prints: OpenID Connect Apache Module (mod_auth_openidc)
mod_auth_openidc-2.4.1

So newer !

**Reply by wkulesza** [reply]
Is it easy to downgrade from 2.4.1 to 2.3.3?

Edit: i was able to update to latest version and indeed the problem with edge/chrome 500 internal server error dissapeared.
